spyder/plugins/ipythonconsole/widgets/shell.py

```diff
@@ -452,6 +453,26 @@ def change_mpl_backend(self, command):
             if not 'inline' in command:
                 self.silent_execute(command)
 
+    # ---- Spyder-kernels methods -------------------------------------------
+    def get_editor(self, filename=None):
```

Comment:
This method is unnecessary, right? If so, please remove it.

Reply:
Yes but not for long. It is used for run_cell